### PR TITLE
Support Style/FileName: ExpectMatchingDefinition acronym

### PR DESCRIPTION
**Pull request by iguchi1124**
Related: https://github.com/bbatsov/rubocop/issues/3793

WDYT?

---

Before submitting the PR make sure the following are checked:

* [ ] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/style/file_name.rb [hunk 1]
```diff
@@ -17,6 +17,14 @@ class FileName < Cop
 
         SNAKE_CASE = /^[\da-z_.?!]+$/
 
+        ACRONYM_NAME_MAPPINGS = {
+          'Cli'  => 'CLI',
+          'Dsl'  => 'DSL',
+          'Http' => 'HTTP',
+          'Rfc'  => 'RFC',
+          'Xml'  => 'XML'
+        }.freeze
```
Comment (pocke) [inline lib/rubocop/cop/style/file_name.rb]: I think this list is incomplete(e.g. `URI` should be included into the list).
Maybe `golint` source code is helpful. https://github.com/golang/lint/blob/master/lint.go#L731-L770

File: lib/rubocop/cop/style/file_name.rb [hunk 2]
```diff
@@ -126,6 +135,15 @@ def match?(expected)
           expected.empty? || expected == [:Object]
         end
 
+        def match_acronym_name?(name, expected)
```
Comment (glennpratt) [inline lib/rubocop/cop/style/file_name.rb]: Arguments are reversed from the usage.

**Comment by glennpratt** [comment]
@iguchi1124 awesome work opening this so fast.

For my use case, we have a truck load of acronyms beyond what is listed. If it were up to me, I would just allow any word from the snake case to be an acronym. I'd be happy to supply code for that if that would be acceptable.

If that's too lax, I'll just have to create a follow-up to allow configuring ACRONYM_NAME_MAPPINGS.

File: lib/rubocop/cop/style/file_name.rb [hunk 3]
```diff
@@ -126,6 +135,15 @@ def match?(expected)
...
+          name = name.to_s
+          expected = expected.to_s
+
+          ACRONYM_NAME_MAPPINGS.any? do |(key, value)|
+            name.gsub(/#{key}/, value) == expected
```
Comment (glennpratt) [inline lib/rubocop/cop/style/file_name.rb]: I think `/#{key}/` can just be `key`, don't see any reason to create a Regexp.

**Comment by bbatsov** [comment]
> If that's too lax, I'll just have to create a follow-up to allow configuring ACRONYM_NAME_MAPPINGS.

Let's go with some configurable mappings. I'd rather not hardcode this.

File: lib/rubocop/cop/style/file_name.rb (hunk 1 above)
Comment (bbatsov) [inline lib/rubocop/cop/style/file_name.rb]: Good tip!

Comment (iguchi1124) [inline lib/rubocop/cop/style/file_name.rb]: @pocke thanks for teach me helpful examples!

Add default allowed acronyms https://github.com/bbatsov/rubocop/pull/3798/commits/1efb5bb69ba8e6c3dd1a6790c50b2baa8793ffa0

**Comment by iguchi1124** [comment]
> Let's go with some configurable mappings. I'd rather not hardcode this.

Fix to configurable allowed acronyms at yaml configurations file. https://github.com/bbatsov/rubocop/pull/3798/commits/ce0c35ddc9fb2062b3301b89602de904c743732d

**Pull request merged**